build.sbt

```diff
@@ -1,30 +1,84 @@
 import java.nio.file.Paths
 
-lazy val `akka-management` = project
+// root
+lazy val `akka-management-root` = project
   .in(file("."))
   .settings(unidocSettings)
   .enablePlugins(NoPublish)
   .disablePlugins(BintrayPlugin)
-  .aggregate(`cluster-http`, docs)
+  .aggregate(
+    `akka-discovery`,
+    `akka-discovery-dns`,
+    `akka-management`,
+    `cluster-http`,
+    `cluster-bootstrap`,
+    docs)
+
+// interfaces and extension for Discovery
+lazy val `akka-discovery` = project
+  .in(file("discovery"))
+  .enablePlugins(AutomateHeaderPlugin)
+  .settings(unidocSettings)
+  .settings(
+    name := "akka-discovery",
+    Dependencies.Discovery
+  )
+
+// DNS implementation of discovery, default and works well for Kubernetes among other things
+lazy val `akka-discovery-dns` = project
+  .in(file("discovery-dns"))
+  .enablePlugins(AutomateHeaderPlugin)
+  .settings(unidocSettings)
+  .settings(
+    name := "akka-discovery-dns",
+    Dependencies.DiscoveryDns
+  )
+  .dependsOn(`akka-discovery`)
+
+// gathers all enabled routes and serves them (HTTP or otherwise)
+lazy val `akka-management` = project
+  .in(file("management"))
+  .enablePlugins(AutomateHeaderPlugin)
+  .settings(unidocSettings)
+  .settings(
+    name := "akka-management",
+    Dependencies.ManagementHttp
+  )
 
+// cluster management http routes, expose information and operations about the cluster
 lazy val `cluster-http` = project
   .in(file("cluster-http"))
   .enablePlugins(AutomateHeaderPlugin)
   .settings(
     name := "akka-management-cluster-http",
-    Dependencies.ClusterHttp,
-    resolvers += Resolver.bintrayRepo("hajile", "maven")
+    Dependencies.ClusterHttp
+  )
+  .dependsOn(`akka-management`)
+
+// cluster bootstraping
+lazy val `cluster-bootstrap` = project
+  .in(file("cluster-bootstrap"))
+  .enablePlugins(AutomateHeaderPlugin)
+  .settings(
+    name := "akka-management-cluster-bootstrap",
+    Dependencies.ClusterBootstrap
   )
+  .dependsOn(`akka-management`, `akka-discovery`)
 
-lazy val `joining-demo` = project
-  .in(file("joining-demo"))
+// TODO cluster-bootstrap-dns which would just pull together things
```

Comment:
I will clarify in there, it's about introducing a module called `akka-management-cluster-bootstrap-dns` which does not do anything, except pull together the 2 modules of `cluster bootstrap` and `akka discovery dns` so it's only 1 dependency you need to pick. I was thinking it would be nice to have "hello world bootstrap in the smallest number of steps" so that would reduce 2 deps into 1.